reference-implementation/to-upstream-wpts/writable-streams/bad-underlying-sinks.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+  self.importScripts('../resources/test-utils.js');
+  self.importScripts('../resources/recording-streams.js');
+}
+
+const error1 = new Error('error1');
+error1.name = 'error1';
+
+promise_test(t => {
+
+  const ws = recordingWritableStream({
+    close() {
+      throw error1;
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  return promise_rejects(t, error1, writer.close(), 'close() promise must reject with the thrown error')
+  .then(() => promise_rejects(t, error1, writer.ready, 'ready promise must reject with the thrown error'))
+  .then(() => {
+    assert_array_equals(ws.events, ['close']);
+  });
+
+}, 'Underlying sink close: throwing method');
```

Comment:
Hmm. For certain types of tests I prefer the description to describe the situation being tested, and the assertions themselves to contain the "must"s. Let me tweak this though since I guess most tests are moving in that direction.